Lib/venv/scripts/nt/activate.bat

```diff
@@ -2,7 +2,7 @@
 
 rem This file is UTF-8 encoded, so we need to update the current code page while executing it
 for /f "tokens=2 delims=:" %%a in ('"%SystemRoot%\System32\chcp.com"') do (
-    set "_OLD_CODEPAGE=%%a"
+    for %%b in ("%%a") do set "_OLD_CODEPAGE=%%~nxb"
```

Comment:
We could also modify `delims` in the first loop to include ".", e.g. `"tokens=2 delims=.:"`. It's still the 2nd token because the period, if present, is at the end of the string.

Reply:
> We could also modify `delims`

Yes, that certainly seems easier to understand than the somewhat obscure `~nx` which is really intended for file extensions and the usage here non-obvious.

Both solutions - `delims` and `~nx` - may not work with some other locale if that happens to use another character than `.` in the message - we're at the whims of Windows developers here, in some sense. Is there any way we can make the solution more general to cope with that?

Reply:
Thanks for the very fast reaction. I verified: for /F "tokens=2 delims=:." %a in ('"C:\WINDOWS\System32\chcp.com"') also works on a Swiss German Windows 10 and solves the problem as well.

The alternative I considered was using the C-API:
`for /f %%a in ('%~dp0python.exe -c "from ctypes import windll; print(windll.kernel32.GetConsoleOutputCP());"') do (set "_OLD_CODEPAGE=%%a")`

This is likely more portable among Windows versions. python.exe should be part of the virtualenv scripts folder. The %~dp0 returns the folder of the script file even if called from another batch file.

Reply:
Call `GetConsoleCP` to get the current input codepage, as chcp.com does. CMD actually uses the output codepage to decode the script, which we overwrite and lose when chcp.com sets the codepage. We could remove the reliance on chcp.com entirely by calling `GetConsoleOutputCP` and `SetConsoleOutputCP` via ctypes.

Reply:
okay. I have done that and also saved and set the Console input.